k-distribution/src/main/scripts/bin/krun

```diff
@@ -478,16 +478,20 @@ else
       fi
     fi
     indirectName="declaredConfigVar_$name"
+    indirectModuleName="declaredConfigVarModule_$name"
     sortName=${!indirectName}
+    moduleName=${!indirectModuleName:-unset}
     if [ -z "${!parser_name+unset}" ]; then
       if [ "$name" = "PGM" ]; then
         if $hasArgv; then
           parser=(kparse --directory "$dir")
         else
           parser=(kparse --directory "$dir" --module "$mainModuleName")
         fi
-      else
+      elif [ -z "${!indirectModuleName+unset}" ]; then
```

Comment:
AFAICT, this check will never pass, because we're always defining `indirectModuleName` above. But I also don't understand the `${!...}` syntax..

Reply:
It's an indirect variable lookup. We're not looking up indirectModuleName, we're expanding it as a string and looking up the variable with the name its value expands to. That variable might not be set if the user didn't specify the parser attribute for a particular configuration variable.